src/matching/Generic_vs_generic.ml

```diff
@@ -517,6 +522,44 @@ let rec m_name a b =
   | G.IdQualified _, _ ->
       fail ()
 
+(* This is just an entry point for m_name_inner, which just ensures that we only
+   ever unpack wildcard imports once, before entering the main recursive loop of
+   m_name_inner.
+   Since we might reach m_name through many entry points (IdQualified,
+   DotAccess, Id, etc), it's easier to intercept all of these "naming-related"
+   matching procedures at a common point. This is where they all meet.
+*)
+and m_name a b =
+  m_name_inner a b >!> fun () tin ->
```

Comment:
Why `>!>` here instead of `>||>` ? Good to explain it.

Reply:
I kind of viewed this as a similar situation to symbolic propagation, where there is some extra information "stored within" the node in question that we are matching. I didn't want to incur this unpacking cost (however much that may be) for a name we were going to match anyway, so I decided to make it a `>!>`. Thoughts?

Reply:
AFAICT we use `>||>` with symbolic propagation.